components/Alert/react/Alert.tsx

```diff
@@ -107,37 +107,34 @@ export const Alert: React.FC<
       : undefined)
 
   const [dismissed, setDismissed] = React.useState(false)
-  const [durationTimeout, setDurationTimeout] = React.useState<
-    number | undefined
-  >(undefined)
+  const durationTimeout = React.useRef<number | undefined>(undefined)
 
-  function clearDurationTimeout() {
+  const clearDurationTimeout = React.useCallback(() => {
     if (durationTimeout) {
```

Comment:
```suggestion
    if (durationTimeout.current) {
```

Otherwise this conditional will always pass then the output of useRef is `{ current: undefined }`

Reply:
Actually, seems you can call `clearTimeout` on undefined, so you can remove the `if` altogether, and it's still fine